File: ios-app/modules/agora-viewer/ios/AgoraViewerView.swift
import ExpoModulesCore
import AgoraRtcKit

class AgoraViewerView: ExpoView {
  var isLocal: Bool = false

  func attachRemoteUid(_ uid: UInt) {
    if isLocal { return }
    guard let engine = sharedAgoraEngine else { return }
    let canvas = AgoraRtcVideoCanvas()
    canvas.uid = uid
    canvas.renderMode = .hidden
    canvas.view = self
    engine.setupRemoteVideo(canvas)
  }

  func attachLocal() {
    guard let engine = sharedAgoraEngine else { return }
    let canvas = AgoraRtcVideoCanvas()
    canvas.uid = 0
    canvas.renderMode = .hidden
    canvas.view = self
    engine.setupLocalVideo(canvas)
  }
}

File: ios-app/modules/agora-viewer/android/src/main/java/expo/modules/agoraviewer/AgoraViewerView.kt
package expo.modules.agoraviewer

import android.content.Context
import android.view.SurfaceView
import android.view.ViewGroup
import expo.modules.kotlin.AppContext
import expo.modules.kotlin.views.ExpoView
import io.agora.rtc2.video.VideoCanvas

class AgoraViewerView(context: Context, appContext: AppContext) : ExpoView(context, appContext) {
  fun attachRemoteUid(uid: Int) {
    removeAllViews()
    val surfaceView = SurfaceView(context)
    addView(
      surfaceView,
      ViewGroup.LayoutParams(
        ViewGroup.LayoutParams.MATCH_PARENT,
        ViewGroup.LayoutParams.MATCH_PARENT,
      ),
    )
    val engine = AgoraViewerModule.sharedEngine ?: return
    val canvas = VideoCanvas(surfaceView, VideoCanvas.RENDER_MODE_HIDDEN, uid)
    engine.setupRemoteVideo(canvas)
  }
}
